### PR TITLE
Configure MathType Web Integrations project to use Cypress Dashboard at project level

### PR DESCRIPTION
**Pull request by carla-at-wiris**
## Description

This Pull Request includes the necessary documentation for this project on the testing integration with Cypress Dashboard, such as:
* When should the tests be sent to the Cypress Dashboard.
* How to send local executions to run on Cypress Dashboard through GitHub actions.

There are also the following changes apart from the documentation:
* An update on the node and Ubuntu versions where the tests should run.
* Change log update.
* Fix a typo on the test:run script.
* Add the Cypress Dashboard project ID to link this repository with the Dashboard.

Outside the code, I've added the slack integration and configurations to the Cypress Dashboard, and also, the project record key and ID have also been added to the project secrets as:
* CYPRESS_RECORD_KEY
* CYPRESS_PROJECT_ID

No ADRs' were needed since all the decisions made are more of the best practices type and have been documented on the integrations-guides private GitHub repository.

All the information related to the analysis of the Cypress Dashboard tool can be found [here](https://docs.google.com/document/d/1RCkDFH7dJWBH0WlbfBAJSNVI3bA1M1nat3--jyEhvOY/edit#)

## Steps to reproduce

Since this is more of a documenting task, there is nothing to reproduce.
You can verify that the information added to the project, about Cypress Dashboard, is correct.

---

[#taskid 17158](https://wiris.kanbanize.com/ctrl_board/2/cards/17158/details/)

### CODE REVIEW

File: .github/workflows/cypress-Run-all-tests-no-cache.yml [hunk 1]
```diff
@@ -69,32 +76,49 @@ jobs:
...
+      #    echo "CY_RECORD_FLAG=" >> $GITHUB_ENV
+
+      # We define the environment variables to send the test execution to the Cypress Dashboard
+      # on every pull request or when specifying to record the tests on workflow dispatch
+      # - name: Set Dashboard flags when pull_request or worflow_dispatch
+      #  if: ${{ github.event_name == 'pull_request' || github.event.inputs.cyDashboard == 'Yes' }}
+      #  run: |
+      #    echo "CY_RECORD_KEY=${{ secrets.CYPRESS_RECORD_KEY }}" >> $GITHUB_ENV
+      #    echo "CY_RECORD_FLAG=-- --record --key " >> $GITHUB_ENV
+
```
Comment (xripoll-at-wiris) [inline .github/workflows/cypress-Run-all-tests-no-cache.yml:97]: Is this code outdated? If so, it's better to just remove it.

File: .github/workflows/cypress-Run-all-tests-no-cache.yml [hunk 2]
```diff
@@ -69,32 +76,49 @@ jobs:
...
             cypress/tests/**/*.js
           # config: "\"video\": true, \"screenshotsFolder\": \"cypress/screenshots/${{ matrix.browser }}\", \"videosFolder\": \"cypress/videos/${{ matrix.browser }}\", \"video\": true"
           # spec: cypress/tests/${{ matrix.type }}/*.js
-        # env:
+        env:
         # https://github.com/cypress-io/cypress-example-kitchensink/settings/secrets
-            # screenshotsFolder: "cypress/screenshots/${{ matrix.browser }}"
-            # videosFolder: "cypress/videos/${{ matrix.browser }}"
+        #     # screenshotsFolder: "cypress/screenshots/${{ matrix.browser }}"
+        #     # videosFolder: "cypress/videos/${{ matrix.browser }}"
```
Comment (xripoll-at-wiris) [inline .github/workflows/cypress-Run-all-tests-no-cache.yml:119]: Watch out for things like this in PRs: if the behavior hasn't changed, try not to commit unnecessary changes (like the double comments).

File: docs/development/testing/README.md [hunk 3]
```diff
@@ -43,4 +43,21 @@ $ npm test
 
 ## Run all the tests for a specific demo
 
-// TODO
\ No newline at end of file
+// TODO
+
+## Record the tests on Cypress Dashboard
```
Comment (dcanet-at-wiris) [inline docs/development/testing/README.md:48]: Since changes have not been included on the workflow, imho It makes no sense to update the development guides.

**Review by dcanet-at-wiris — CHANGES_REQUESTED** [review]
### Use environment variables and CI/CD secrets for all related Cypress values

I would use environment variables for both project_id and record_key, 

This would entail the creation of secrets at repository level.

They would be named CYPRESS_PROJECT_ID and CYPRESS_RECORD_KEY to follow the standard.

I am considering that these secrets will not be documented anywhere, except the workflows code, and we may need to document their existence some where. At this (public) repo or at an another (private) repo. 

More info: https://docs.cypress.io/guides/guides/environment-variables

### Included files not related to the PR

These files shouldn't be included on this PR since changes haven't been applied. 

I wouldn't add commented code, as a rule of thumb.

- demos/html5/generic/package-lock.json
- .github/workflows/cypress-Run-all-tests-no-cache.yml
- .github/workflows/cypress-Run-all-tests-with-cache.yml

File: .github/workflows/cypress-Run-all-tests-no-cache.yml (hunk 2 above)
Comment (dcanet-at-wiris) [reply .github/workflows/cypress-Run-all-tests-no-cache.yml:119]: > This Pull Request includes the necessary documentation for this project on the testing integration with Cypress Dashboard, such as:
> 
>     * When should the tests be sent to the Cypress Dashboard.
This Pull Request includes the necessary documentation for this project on the testing integration with Cypress Dashboard, such as:

>    When should the tests be sent to the Cypress Dashboard.
>    How to send local executions to run on Cypress Dashboard through GitHub actions.

These goals above are out of the scope of this task and therefore they shouldn't be considered on this pull request.

**Comment by dcanet-at-wiris** [comment]
> All the information related to the analysis of the Cypress Dashboard tool can be found here

Since this is a document of private use of the dev team I wouldn't reference it here for transparency sake.

File: cypress.json [hunk 4]
```diff
@@ -2,10 +2,10 @@
     "baseUrl": "http://localhost:8007",
     "integrationFolder": "cypress/tests",
     "screenshotOnRunFailure": true,
+    "projectId": "efpemq",
```
Comment (dcanet-at-wiris) [inline cypress.json:5]: I would use an environment variable here.

First, Imho it's much safer and it follows the Cypress recommendations. 

Second, It will be easier to maintain, in case the projectId value is used elsewhere on the project: ci/cd workflows, documentation, ...

Third, using an environement variable will allow us to assign a value on our local computer and get the value from a secret key on the CI/CD environment.

Comment (dcanet-at-wiris) [reply cypress.json:5]: See: https://docs.cypress.io/guides/guides/environment-variables

**Review by dcanet-at-wiris — CHANGES_REQUESTED** [review]
This comment from @xripoll-at-wiris  has not been considered [Watch out for things like this in PRs: if the behavior hasn't changed, try not to commit unnecessary changes (like the double comments)](https://github.com/wiris/html-integrations/pull/474#discussion_r760010786).

Therefore, these next files shouldn't be added to the PR, if you believe they no functional changes need to be made to them: 

- .github/workflows/cypress-Run-all-tests-no-cache.yml
- package.json
- cypress.json


But also...

## The cypress.json file has no change at all now and 'projectId' is not used anymore. 🤔 

  On our last review, and also [on this comment](https://github.com/wiris/html-integrations/pull/474#pullrequestreview-820160089), I believe we agreed to add the CYPRESS_PROJECT_ID tag and using an environment variable for the value.

  On this version you've removed the parameter at all. Could you explain why you changed your mind about adding this value or not.

  Is it better not to add the [projectId option to the cypress.json file](https://docs.cypress.io/faq/questions/using-cypress-faq#What-is-the-projectId-for) at all? or do [we use the 'env' option](https://docs.cypress.io/guides/references/configuration#Options) of the cypress.json file?



These next files shouldn't include any change since they are not involved on the PR itself

File: docs/development/testing/README.md (hunk 3 above)
Comment (dcanet-at-wiris) [reply docs/development/testing/README.md:48]: @carla-at-wiris ?

Comment (carla-at-wiris) [reply docs/development/testing/README.md:48]: It makes sense, then this would mean to close this PR since this is the only change made at code level

Comment (dcanet-at-wiris) [reply docs/development/testing/README.md:48]: No, I didn't mean that.

**Comment by dcanet-at-wiris** [comment]
After today's project review we decided to close this PR
